### PR TITLE
#2321: Add page and block template workarounds for empty sidebars issue affecting recent Menu Block releases.

### PR DESCRIPTION
**Pull request by joeparsons**
*Note this PR targets the dependabot branch for updating menu_block to the latest version (and should hopefully un-block that PR).*

This PR adds a workaround to the page template (inspired by https://www.drupal.org/project/menu_block/issues/3271218#comment-14499729) to deal with the empty sidebar issue that's prevented us from updating to the latest Menu Block release(s).

## Related issues

Provides workaround for #2321

## How to test

Shared by @bberndt-uaz in #2320:
>Tested in Probo just in case: I created a new system menu, added the menu as a block to the Sidebar First region, and disabled the default Quickstart sidebar menu block. With the Initial visibility level set to 1, the new sidebar block appears on all pages. With the Initial visibility level set to 2, the sidebar properly appears only on the page which is included in the new menu.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by bberndt-uaz — COMMENTED** [review]
Tested in Probo using the same steps as in the How to Test section in the PR description above. The test menu in the left sidebar (sidebar_first region) is displayed properly with the initial visibility set to 1 or 2. However, on the [Calendar page](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2453.probo.build/calendar) this PR appears to remove the calendar from the right sidebar (sidebar_second region).

File: themes/custom/az_barrio/templates/layout/page.html.twig [hunk 1]
```diff
@@ -200,14 +200,16 @@
               <main{{ content_attributes }}>
                 {{ page.content }}
               </main>
-              {% if page.sidebar_first %}
+              {% set has_sidebar_first = page.sidebar_first|render|striptags|trim is not empty %}
```
Comment (joeparsons) [inline themes/custom/az_barrio/templates/layout/page.html.twig:203]:
```suggestion
              {% set has_sidebar_first = page.sidebar_first|render|striptags('<drupal-render-placeholder><embed><hr><iframe><img><input><link><object><script><source><style><video>')|trim is not empty %}        
```

File: themes/custom/az_barrio/templates/layout/page.html.twig [hunk 2]
```diff
@@ -200,14 +200,16 @@
...
+              {% if has_sidebar_first %}
               <div{{ sidebar_first_attributes }}>
                 <aside class="section" role="complementary">
                   {{ page.sidebar_first }}
                 </aside>
               </div>
               {% endif %}
-              {% if page.sidebar_second %}
+              {% set has_sidebar_second = page.sidebar_second|render|striptags|trim is not empty %}
```
Comment (joeparsons) [inline themes/custom/az_barrio/templates/layout/page.html.twig:211]:
```suggestion
              {% set has_sidebar_second = page.sidebar_second|render|striptags('<drupal-render-placeholder><embed><hr><iframe><img><input><link><object><script><source><style><video>')|trim is not empty %}      
```

**Comment by joeparsons** [comment]
Updated workaround in page template to allow certain tags `striptags`.  Copied list of tags to allow from this module:
https://www.drupal.org/project/twig_real_content

**Comment by trackleft** [comment]
The alternative to this approach would be to require that site administrators convert all menu blocks to quickstart menu blocks.

**Comment by bberndt-uaz** [comment]
Confirmed that the menu block is not displayed when the **Display title** checkbox is unchecked, the menu is empty, and the initial visibility of the menu block is set to 1 (always visible).

**Comment by joeparsons** [comment]
Can't add an actual approval since I originally created this PR but i approve the latest changes made by @trackleft and think we should merge this ✅

**Review by trackleft — APPROVED** [review]

**Pull request merged**